test/plugins/individual-collateral/angle/constants.ts

```diff
@@ -0,0 +1,17 @@
+import { bn, fp } from '../../../../common/numbers'
+import { networkConfig } from '../../../../common/configuration'
+
+// Mainnet Addresses
+export const USDA = networkConfig['31337'].tokens.USDA as string
+export const StUSD = networkConfig['31337'].tokens.stUSD as string
+export const USDA_USD_PRICE_FEED = networkConfig['31337'].chainlinkFeeds.USDC as string // we use USDC feed as USDA/USD is not available in chainlink, and USDA/USDC is always 1 through the transmuter
```

Comment:
ok this is a critical piece of the pricing puzzle so i think its worth justifying this with a more in-depth explanation.
can you link the transmuter docs and code?
is USDC **ALWAYS instantly and atomically exchangeable 1:1** for USDA?
are there any failure modes?

Reply:
Please find here the Transmuter [code](https://github.com/AngleProtocol/angle-transmuter) and [docs](https://docs.angle.money/transmuter/transmuter).
USDA is in the current settings exchangeable 1:1 instantly and atomically with USDC.
As you point out, there are some failure modes (and this is where the value of the Transmuter lies), especially if USDC was to depeg. Transmuter was designed to incorporate anti-depeg mechanisms, and automatically fallback into a preservation mode if one of the assets in its backing was to depeg. As currently all assets for which this monitoring is activated are USDC derivatives (namely USDC and steakUSDC on Morpho) then, only a USDC depeg could cause the 1:1 exchangeability to break between the 2 assets.
There are some cases where the protocol could have its liquidity buffer in USDC depleted, but in this case, the system supports a permissionless rebalancing feature allowing anyone to refill the USDC reserves of the protocol and enable 1:1 USDA->USDC conversion.

Reply:
i think having the liquidity buffer depleted is a concern. there's really no guarantees on when it would be refilled, right? and at that point, users would need to go to the market to exchange?

Reply:
In the current setup with the public harvester contract, and the liquidity of the collateral assets, there is in fact a guarantee that this liquidity buffer will be refilled.
While we mentioned the liquidity buffer in USDC, with the transmuter, users can burn their USDA for any of the other assets in the backing (which is not the case by the way with the majority of other stablecoin protocols), and so before selling on the market, they can burn for like steakUSDC and USDM. They may also choose to redeem their USDA for a portion of all the assets in the backing depending on the proportion that they represent.
In any case, users are never locked in Angle and can always exit using the system's primary minting mechanism